code/datums/observation/observation.dm

```diff
@@ -236,3 +236,113 @@
 					unregister(source, listener, proc_call)
 
 	return TRUE
+
+/**
+ * Register to listen for a observation from the passed in listener
+ *
+ * This sets up a listening relationship such that when the listener object emits a observation
+ * the source datum this proc is called upon, will recieve a callback to the given proctype
+ * Return values from procs registered must be a bitfield
+ *
+ * Arguments:
+ * * datum/listener The listener to listen for observations from
+ * * obs_type_or_types Either a string observation name, or a list of observation names (strings)
+ * * proctype The proc to call back when the observation is emitted
+ * * override If a previous registration exists you must explicitly set this
+ */
+/datum/proc/RegisterObservation(datum/listener, obs_type_or_types, proctype, override = FALSE)
```

Comment:
as above, leave this to the singletons, they can just put data on the source datum as needed